src/ng/directive/validators.js

```diff
@@ -62,20 +62,26 @@
  *   </file>
  * </example>
  */
-var requiredDirective = function() {
+var requiredDirective = function($parse) {
   return {
     restrict: 'A',
     require: '?ngModel',
     link: function(scope, elm, attr, ctrl) {
       if (!ctrl) return;
+      var oldVal = attr.required || $parse(attr.ngRequired)(scope);
```

Comment:
What if `attr.required` is `false`? 🤔

Reply:
I believe that `attr.required === false` can't occur, right? If the attribute exists at all then, being a boolean attribute, its value will be `true`.